Assertion failing in Ember beta/canary attempted to remove a function listener

- [x] I am on the [latest](https://github.com/ember-intl/ember-intl/releases/latest) ember-intl version
- [x] I have searched the [issues](https://github.com/ember-intl/ember-intl/issues) of this repo and believe that this is not a duplicate

Tests in my addon are failing against beta/canary with the message
`Error: Assertion Failed: You attempted to remove a function listener which did not exist on the instance, which means you may have attempted to remove it before it was added.` wheverever the `{{t}}` helper is used. Since the same tests are failing on `master` here I didn't dig too much further just opening this issue for discussion and for others to find.

### Steps to Reproduce

Run `ember-intl` tests against beta/canary for example: 
https://travis-ci.org/ember-intl/ember-intl/jobs/639709922#L605

@jasonmit right now I'm getting
```
Cannot find module '@ember-data/canary-features/addon/default-features.js'
Require stack:
- /Users/kyle.turney/dev/ember-intl/node_modules/@ember-data/-build-infra/src/features.js
- /Users/kyle.turney/dev/ember-intl/node_modules/@ember-data/-build-infra/src/debug-macros.js
- /Users/kyle.turney/dev/ember-intl/node_modules/@ember-data/-build-infra/src/addon-build-config-for-data-package.js
- /Users/kyle.turney/dev/ember-intl/node_modules/ember-data/index.js
- /Users/kyle.turney/dev/ember-intl/node_modules/ember-cli/lib/models/package-info-cache/package-info.js
- /Users/kyle.turney/dev/ember-intl/node_modules/ember-cli/lib/models/package-info-cache/index.js
- /Users/kyle.turney/dev/ember-intl/node_modules/ember-cli/lib/models/project.js
- /Users/kyle.turney/dev/ember-intl/node_modules/ember-cli/lib/utilities/get-config.js
- /Users/kyle.turney/dev/ember-intl/node_modules/ember-cli/lib/utilities/instrumentation.js
- /Users/kyle.turney/dev/ember-intl/node_modules/ember-cli/lib/cli/index.js
- /Users/kyle.turney/dev/ember-intl/node_modules/ember-cli/bin/ember
```
while trying to run the ember-canary scenario locally.
Would bringing https://github.com/ember-intl/ember-intl/pull/1042 into the 4.x branch fix that?

--- a/addon/helpers/-format-base.js
+++ b/addon/helpers/-format-base.js
@@ -19,7 +19,7 @@ const AbstractHelper = Helper.extend({
     this._super();
 
     this.intl = getOwner(this).lookup('service:intl');
-    this.intl.on('localeChanged', this, this.recompute);
+    this.intl.on('localeChanged', this, 'recompute');
   },
 
   format() {
@@ -43,7 +43,7 @@ const AbstractHelper = Helper.extend({
   willDestroy() {
     this._super();
 
-    this.intl.off('localeChanged', this, this.recompute);
+    this.intl.off('localeChanged', this, 'recompute');
   }
 });
 

--- a/config/ember-try.js
+++ b/config/ember-try.js
@@ -11,6 +11,7 @@ module.exports = function() {
           name: 'ember-lts-2.12',
           npm: {
             devDependencies: {
+              'ember-data': '~3.12.0',
               'ember-source': '~2.12.0',
               '@ember/jquery': '^0.5.2'
             }

--- a/package.json
+++ b/package.json
@@ -80,7 +80,6 @@
     "ember-cli-shims": "^1.2.0",
     "ember-cli-testdouble-qunit": "^2.1.1",
     "ember-code-snippet": "^3.0.0",
-    "ember-data": "~3.12.0",
     "ember-disable-prototype-extensions": "^1.1.3",
     "ember-export-application-global": "^2.0.0",
     "ember-load-initializers": "^2.0.0",

